Zend/zend_object_handlers.c

```diff
@@ -905,7 +905,13 @@ ZEND_API zval *zend_std_read_property(zend_object *zobj, zend_string *name, int
 			if (zobj->ce->__get && !((*guard) & IN_GET)) {
 				goto call_getter;
 			}
+
+			bool obj_is_freed = GC_REFCOUNT(zobj) == 1;
 			OBJ_RELEASE(zobj);
+			if (UNEXPECTED(obj_is_freed)) {
+				retval = &EG(uninitialized_zval);
```

Comment:
This reverts the return value of __isset() from ``true`` to ``false``.

Reply:
I don't see a big danger (just made a note).
I don't object against this fix (it looks good to me).
@nielsdos please also take a quick look.

Reply:
I kinda liked the original approach more because the fix was more "localized". I also don't think it's a big danger, especially because the test is pathological code.
Anyway, one question I had was with the `obj_is_freed` condition. What if the object is cyclic, and the refcount>1, but after the OBJ_RELEASE call the object is released due to its cycles anyway: is this possible?
I tried to cheat it, but didn't manage to make it crash.

Reply:
I also like the original approach.

I think we don't need to set `retval` here, as it is initialized to this value [above](https://github.com/php/php-src/blob/6c2e97b466ce80ef072209a663840b72e7817646/Zend/zend_object_handlers.c#L883), and is not changed after that? The result of `__isset` is only used to decide whether to call `__get`, and is not exposed. `$c->prop ?? null` is always null when the object implements `__isset` but not `__get`.

I'm not sure that falling-through when `__isset` returns true was on purpose. I think we can `goto exit` in all cases.

Reply:
I believe the existing behavior was not on purpose / is a bug.

We normally do not initialize lazy objects when there is a magic method to carry the operation. The reason is that we assume that if the magic method depends on the object's state or observes it, its execution will trigger initialization as needed.

In this particular case we call `__isset()` followed by `__get()` (if `__isset()` returned true). But when `__get()` is not defined, we always return `null`. Therefore, the object's state is not used after the `__isset()` call, so we don't need initialization.